### PR TITLE
Blagoev/promise3

### PR DESCRIPTION
**Pull request by blagoev**
this is continuation of the effort to provide Promisified API for auth methods
old PR here: https://github.com/realm/realm-js/pull/1249

### CODE REVIEW

**Comment by blagoev** [comment]
For review: Note this: https://github.com/realm/realm-js/pull/1302/commits/115a2c99023a30060145f88040d1c1a2a25c29fb#diff-4d6841a5998a2db16e22741bc65e4f98R193

We output a warning on every use. Not that I like that too much. It is the same for Realm.openAsync.
What's your input on that. Repeal and Replace ?!?

File: docs/sync.js [hunk 1]
```diff
@@ -148,9 +149,10 @@ class User {
      * @param {string} options.provider - The provider type
      * @param {string} options.providerToken - The access token for the given provider
      * @param {object} [options.userInfo] - A map containing additional data required by the provider
-     * @param {function(error, User)} callback - called with the following arguments:
+     * @param {function(error, User)} [callback] - an optional called with the following arguments:
```
Comment (kneth) [inline docs/sync.js]: "optional" -> "optional function"???

**Comment by kneth** [comment]
Looks good to me.

File: docs/sync.js (hunk 1 above)
Comment (blagoev) [reply docs/sync.js]: that got blindly copied from the other PR, can reword it

**Pull request merged**